controller/api/destination/external-workload/endpoints_controller.go

```diff
@@ -437,15 +437,15 @@ func (ec *EndpointsController) onAddExternalWorkload(obj interface{}) {
 		return
 	}
 
-	for _, svc := range services {
+	for svc := range services {
 		ec.queue.Add(svc)
 	}
 }
 
 func (ec *EndpointsController) onUpdateExternalWorkload(old, cur interface{}) {
 	services := ec.getServicesToUpdateOnExternalWorkloadChange(old, cur)
 
-	for _, svc := range services {
+	for svc := range services {
```

Comment:
This was resulting in a panic whenever these callbacks fire:


```
time="2024-01-24T08:12:27Z" level=info msg="waiting for caches to sync"
time="2024-01-24T08:12:27Z" level=info msg="caches synced"
time="2024-01-24T08:12:27Z" level=info msg="waiting for caches to sync"
time="2024-01-24T08:12:27Z" level=info msg="caches synced"
time="2024-01-24T08:12:27Z" level=info msg="waiting for caches to sync"
time="2024-01-24T08:12:27Z" level=info msg="caches synced"
time="2024-01-24T08:12:27Z" level=info msg="starting gRPC server on :8086"
panic: interface conversion: interface {} is sets.Empty, not string

goroutine 287 [running]:
github.com/linkerd/linkerd2/controller/api/destination/external-workload.(*EndpointsController).processQueue(0xc000274790)
        /linkerd-build/controller/api/destination/external-workload/endpoints_controller.go:235 +0x105
created by github.com/linkerd/linkerd2/controller/api/destination/external-workload.(*EndpointsController).Start in goroutine 1
        /linkerd-build/controller/api/destination/external-workload/endpoints_controller.go:218 +0x125
```

Again, we need to resort to functional testing quite a bit here

Reply:
Nice, I managed to catch this separately when I fired up the controller. Fixed and confirmed it works.